### PR TITLE
Refactor to Commando and Typescript

### PR DESCRIPTION
**Pull request by kevinzwang**
- now uses Discord.js Commando instead of jank (but admittedly pretty feature rich) custom commands solution
- also changed everything to typescript <3
- major refactoring, including moving lots of functionality out of `bot.js` and into more separate files such as `verification.js`
- removed `classtorole` command and relevant code because it was too esoteric to refactor. I've provided unimplemented commands and event listeners in case you would like to add this back in the future
- env file takes role IDs instead of names

### CODE REVIEW

**Comment by jenchanws** [comment]
I think you should add `dist` to `.gitignore` too and remove them from this PR

**Comment by kevinzwang** [comment]
oh shoot great catch @jenchanws

File: README.md [hunk 1]
```diff
@@ -11,8 +11,8 @@ Before running, configure the test bot by adding an .env file with the following
 PREFIX=>
 TOKEN=[token]
 GUILD_ID=[guild id]
-VERIFIED_ROLE_NAME=[verified role name]
-ADMIN_ROLE_NAME=[admin role name]
+VERIFIED_ROLE_ID=[verified role name]
+ADMIN_ROLE_ID=[admin role name]
```
Comment (jenchanws) [inline README.md:15]: Why was this changed to ID? Now you'd have to figure out the role number instead of just using the name.

File: package.json [hunk 2]
```diff
@@ -19,11 +20,18 @@
   "homepage": "https://github.com/feefs/discordbot#readme",
   "dependencies": {
     "discord.js": "^12.2.0",
+    "discord.js-commando": "github:discordjs/Commando",
```
Comment (jenchanws) [inline package.json:23]: Is there no suitable version on npm?

File: tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+            "*": [
+                "node_modules/*"
+            ]
+        },
+        "strictNullChecks": false
+    },
+    "include": [
+        "src/**/*"
+    ],
+    "exclude": []
```
Comment (jenchanws) [inline tsconfig.json:20]: Is this line necessary?

File: src/commands/verify.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    }
+    
+    execute(message: CommandoMessage) {
+        return message.say(
+            new MessageEmbed({
+                title: 'Verification',
+                description: 'Please submit your Berkeley email for verification.\n' +
+                    'Your information will not be stored or tied to your Discord account.',
+                color: 0xfdb515
+            }).addField('Instructions:', 'Submit your email using\n\`>email [berkeley.edu email]\`\n\n')
```
Comment (jenchanws) [inline src/commands/verify.ts:24]: Use the config prefix here instead of hardcoding >

File: src/commands/poll.ts [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+            description: contents,
+            color: 0xfdb515,
+        }).setAuthor(
+            message.author.username, 
+            message.author.avatarURL({ dynamic: true })
+        )
+
+        if (!/\{|}/.test(message.content)) {
+            message.delete()
+            await message.say('> Items for `>poll` should be in curly brackets {}')
```
Comment (jenchanws) [inline src/commands/poll.ts:50]: Use the config prefix here instead of hardcoding >

File: README.md (hunk 1 above)
Comment (kevinzwang) [reply README.md:15]: this is to prevent roles with the same name causing problems. a guild id is already needed, so you'll need to go into developer mode anyway. role IDs would not be that much harder to get

File: package.json (hunk 2 above)
Comment (kevinzwang) [reply package.json:23]: nope. this is the version that is compatible with discord.js v12

File: tsconfig.json (hunk 3 above)
Comment (kevinzwang) [reply tsconfig.json:20]: yes it is. it's needed for typescript transpilation to work

File: src/commands/verify.ts (hunk 4 above)
Comment (kevinzwang) [reply src/commands/verify.ts:24]: true. i will fix

File: package.json (hunk 2 above)
Comment (jenchanws) [reply package.json:23]: :pepehands:

**Comment by kevinzwang** [comment]
btw once it is approved let me know so that i can do a rebase to combine all these loose commits into one

File: src/commands/poll.ts [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+            description: contents,
+            color: 0xfdb515,
+        }).setAuthor(
+            message.author.username, 
+            message.author.avatarURL({ dynamic: true })
+        )
+
+        await message.say(embed)
+        await message.say(`> Your input: \`${message.content}\``)
+        const confirmMessage = await message.say('> Does this look good?') as Message
```
Comment (feefs) [inline src/commands/poll.ts:55]: can you please send these three to the author's dms?

File: src/commands/poll.ts [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+                'poll {which is the superior cs?} {EECS} {LSCS} {cogsci} {cool socks}'
+            ]
+        })
+    }
+
+    async execute(message: CommandoMessage, args: string) {
+        // i did not bother to look through this code and clean it up other than small things but it seems to work
+        const pollItems = args.split(/\s+(?={)/)
+        if (pollItems.length < 2) {
+            return message.say(
```
Comment (feefs) [inline src/commands/poll.ts:27]: Can this be direct as well?

**Comment by kevinzwang** [comment]
alright it is time for the rebase

**Comment by kevinzwang** [comment]
actually you should just "squash and merge" cuz the merged commit history is linear. @feefs
